src/shaders/_prelude_fog.vertex.glsl

```diff
@@ -1,3 +1,15 @@
 #ifdef FOG
 
-#endif
\ No newline at end of file
+uniform mat4 u_cam_matrix;
+
+vec3 fog_position(vec3 pos) {
+    vec4 p = u_cam_matrix * vec4(pos, 1);
+    return p.xyz / p.w;
```

Comment:
Are you positive we need `/ p.w`? I thought u_cam_matrix would ensure w === 1.

Reply:
Good point. I guess as long as it's strictly affine, that is the case. I try to at least add a comment noting it's a stated assumption.